src/commonTest/kotlin/org/snakeyaml/engine/test_suite/SuiteUtils.kt

```diff
@@ -0,0 +1,114 @@
+package org.snakeyaml.engine.test_suite
+
+import org.snakeyaml.engine.v2.api.LoadSettings
+import org.snakeyaml.engine.v2.api.lowlevel.Parse
+import org.snakeyaml.engine.v2.exceptions.YamlEngineException
+
+internal object SuiteUtils {
+
+    fun parseData(data: YamlTestData): ParseResult {
+        val settings = LoadSettings.builder().setLabel(data.label).build()
+
+        return try {
+            ParseResult(Parse(settings).parseString(data.inYaml).toList())
+        } catch (e: YamlEngineException) {
+            ParseResult(emptyList(), e)
+        }
+    }
+
+    /**
+     * Identifiers of tests that should fail.
+     */
+    val deviationsWithSuccess: Set<YamlTestData.Id> = setOf(
+        //region copied from SnakeYAML
+        // https://github.com/snakeyaml/snakeyaml-engine/blob/5a35e1fe7f780d1405d5a03470f9f13d32b1638a/src/test/java/org/snakeyaml/engine/usecases/external_test_suite/SuiteUtils.java#L33-L34
+        "9C9N",
+        "9JBA",
+        "CVW2",
+        "QB6E",
+        "SU5Z",
+        //endregion
+
+        //region Additional cases
+        "DK95:01",     // https://matrix.yaml.info/details/DK95:01.html
+        "JEF9:02",     // https://matrix.yaml.info/details/JEF9:02.html
+        "L24T:01",     // https://matrix.yaml.info/details/L24T:01.html
+        //endregion
+    ).mapToYamlTestDataId()
+
+    /**
+     * Identifiers of tests that fail with an error
```

Comment:
I'd expect the test suite to provide info which test cases should cause an error. Could you shed more light on it?

Reply:
Good point, the comment is unclear. I'll update it.

The test cases do indeed indicate which are expected to fail and pass, but only one library passes _all_ cases as expected.

https://matrix.yaml.info/

The other libraries will manually override the expected case outcome.